src/otx/core/data/dataset/segmentation.py

```diff
@@ -185,6 +183,17 @@ def __init__(
         )
         self.ignore_index = ignore_index
 
+    @property
+    def has_polygons(self) -> bool:
+        """Check if the dataset has polygons in annotations."""
+        has_polygons = False
+        for subset in self.dm_subset.subsets().values():
+            annot_types = set(subset.get_annotated_type())
+            if annot_types & {"polygon", "ellipse"}:
+                has_polygons = True
```

Comment:
```suggestion
                return True
```
If one of subset has polygons, we can skip later part, I think.